docs/syntax.md

```diff
@@ -123,11 +123,11 @@ This makes the defined function or type visible in other files when the file is
 
 An argument definition of a function, method or class is a type followed by a simple identifier.
 
-A type is a non-union type or multiple `|`-separated non-union types. A non-union type can be:
+A type is a possibly parenthesized non-union type or multiple `|`-separated non-union types.
```

Comment:
I think this should instead be specific to `typedef`. If you need a huge parenthesized type, you should use `typedef` anyway.